app/models/scrum_board.rb

```diff
@@ -0,0 +1,194 @@
+class ScrumBoard < ApplicationRecord
+  has_many :scrum_sprints, -> { order(ended_on: :desc) }, dependent: :destroy,
+                                                          inverse_of: :scrum_board
+  has_many :wish_heaps, -> { order(trello_pos: :desc) }, dependent: :destroy,
+                                                         inverse_of: :scrum_board
+  has_one :scrum_backlog, dependent: :destroy
+
+  alias_attribute :sprints, :scrum_sprints
+  alias_attribute :backlog, :scrum_backlog
+
+  DEFAULT_SPRINT_DURATION = 2.weeks
+  NUM_SPRINTS_FOR_AVG_VELOCITY = 3
+
+  validates :name, presence: true
+  validates :trello_board_id, presence: true
+  validates :trello_url, presence: true
+  validate :trello_url_is_valid
+
+  # Class Methods
+  def self.by_trello_board_or_create(trello_board)
+    scrum_board = ScrumBoard.find_by(trello_board_id: trello_board.id)
+
+    if scrum_board
+      scrum_board.update_from_trello_board(trello_board)
+    else
+      scrum_board = ScrumBoard.create_from_trello_board(trello_board)
+    end
+
+    scrum_board
+  end
+
+  def self.create_from_trello_board(trello_board)
+    scrum_board = ScrumBoard.new(trello_board_id: trello_board.id,
+                                 trello_url: trello_board.url,
+                                 name: trello_board.name,
+                                 last_board_activity_at: trello_board.last_activity_date,
+                                 last_pulled_at: Time.now.utc)
+    scrum_board.save!
+    scrum_board.update_queues_from_trello_board(trello_board)
+    scrum_board.recompute_sprint_metrics
+    scrum_board
+  end
+
+  def self.scrummy_trello_board?(trello_board)
+    # A scrummy board will contain these lists: wish heap, backlog, current
+    scrummy_list_names = ['wish heap', 'backlog', 'current']
+    board_list_names = trello_board.lists.map { |list| list.name.downcase.strip }
+
+    scrummy_list_names.each do |required_name|
+      return false unless board_list_names.any? { |list_name| list_name.include? required_name }
+    end
+
+    true
+  end
+
+  def self.sprinty_trello_list?(trello_list)
+    trello_list.name.downcase.include? 'complete'
+  end
+
+  def self.wishy_trello_list?(trello_list)
+    trello_list.name.downcase.include? 'wish'
+  end
+
+  def self.backloggy_trello_list?(trello_list)
+    trello_list.name.downcase.include? 'backlog'
+  end
+
+  #
+  # Instance Methods
+  #
+  def wish_heap
+    wish_heaps.first
+  end
+
+  def sprint_backlog
+    # This is the "current sprint" list in Trello Board.
+    needle = 'current'
+    trello_list = trello_board.lists.find { |list| list.name.downcase.include? needle }
+    ScrumSprint.sprint_backlog_from_trello_list(self, trello_list)
+  end
+
+  # rubocop: disable Style/SafeNavigation
+  # Simpler just to check if sprint_completed than to try to use SafeNav &. syntax.
+  def current_sprint
```

Comment:
This could conceivably be a new plain-old-ruby-object (PORO) model. Maybe in the future.